distributed/joblib.py

```diff
@@ -45,31 +46,85 @@ def joblib_funcname(x):
     return funcname(x)
 
 
+class Batch(object):
+    def __init__(self, tasks):
+        self.tasks = tasks
+
+    def __call__(self, *data):
+        results = []
+        for func, args, kwargs in self.tasks:
+            args = [a(data) if isinstance(a, itemgetter) else a
+                    for a in args]
+            kwargs = {k: v(data) if isinstance(v, itemgetter) else v
+                      for (k, v) in kwargs.items()}
+            results.append(func(*args, **kwargs))
+        return results
+
+    def __reduce__(self):
+        return (Batch, (self.tasks,))
+
+
 class DaskDistributedBackend(ParallelBackendBase, AutoBatchingMixin):
-    MIN_IDEAL_BATCH_DURATION = 0.2
-    MAX_IDEAL_BATCH_DURATION = 1.0
+    MIN_IDEAL_BATCH_DURATION = 0.5
+    MAX_IDEAL_BATCH_DURATION = 5.0
 
-    def __init__(self, scheduler_host='127.0.0.1:8786', loop=None):
+    def __init__(self, scheduler_host='127.0.0.1:8786', scatter=None, loop=None):
         self.client = Client(scheduler_host, loop=loop)
+        if scatter is not None:
+            # Keep a reference to the scattered data to keep the ids the same
+            self._scatter = list(scatter)
```

Comment:
Then it's bad. We should only accept lists here because technically we can pre-scatter any object so it'd be hard to know if someone wanted us to scatter the object or elements inside it. Will fix to error if not list/tuple.